aidial_adapter_openai/gpt4_multi_modal/download.py

```diff
@@ -131,27 +154,23 @@ async def transform_message(
 async def transform_messages(
     file_storage: Optional[FileStorage], messages: List[dict]
 ) -> Tuple[List[dict], int] | str:
-    new_messages: List[dict] = []
     image_tokens = 0
+    new_messages: List[dict] = []
+    errors: Set[ImageFail] = set()
 
-    errors: Dict[int, List[Tuple[int, str]]] = {}
-
-    n = len(messages)
-    for idx, message in enumerate(messages):
+    for message in messages:
         result = await transform_message(file_storage, message)
         if isinstance(result, list):
-            errors[n - idx] = result
+            errors.update(result)
         else:
             new_message, tokens = result
             new_messages.append(new_message)
             image_tokens += tokens
 
     if errors:
-        msg = "Some of the image attachments failed to download:"
-        for i, error in errors.items():
-            msg += f"\n- {format_ordinal(i)} message from end:"
```

Comment:
minor:

I think that's pretty handful feature, especially for chat users, who will see this as an error stage, to point exact message, where problematic attachment was

Reply:
The problematic attachment in most of the cases will be attached to the **last message**, because we do not allow to continue conversation with an incorrect attachment. So if something went wrong, it's more than likely the problem is the last message.

The exception is when one has changed one model to another. But that should be rare.